app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,112 @@
+class ManageIQ::Providers::IbmCloud::VPC::CloudManager::MetricsCapture < ManageIQ::Providers::CloudManager::MetricsCapture
+  delegate :ext_management_system, :to => :target
+
+  VIM_STYLE_COUNTERS = {
+    "cpu_usage_rate_average"     => {
+      :counter_key           => "cpu_usage_rate_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 1,
+      :rollup                => "average",
+      :unit_key              => "percent",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "disk_usage_rate_average"    => {
+      :counter_key           => "disk_usage_rate_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 2,
+      :rollup                => "average",
+      :unit_key              => "kilobytespersecond",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "net_usage_rate_average"     => {
+      :counter_key           => "net_usage_rate_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 2,
+      :rollup                => "average",
+      :unit_key              => "kilobytespersecond",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "mem_usage_absolute_average" => {
+      :counter_key           => "mem_usage_absolute_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 1,
+      :rollup                => "average",
+      :unit_key              => "percent",
+      :capture_interval_name => "realtime",
+    }.freeze,
+  }.freeze
+
+  def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
+    require 'rest-client'
+
+    raise _("No EMS defined") if ext_management_system.nil?
+
+    initialize_metrics_query(target.name)
+    counters_by_mor = {target.ems_ref => VIM_STYLE_COUNTERS}
+    counter_values_by_mor = {target.ems_ref => {}}
+
+    metrics_endpoint = ext_management_system.endpoints.find_by(:role => "metrics")
+    instance_id = metrics_endpoint.options["monitoring_instance_id"] unless metrics_endpoint.nil?
```

Comment:
We probably should raise if `instance_id` is missing rather than letting the subsequent API call fail

Reply:
I agree it would be better to raise earlier as opposed to letting the API call fail and then catch the exception in https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/308/files#diff-33defe66ca0940649be5af9d6676637d204a884b6016c5f39a736e2322c315c6R75

Would `raise StandardError` be a good option here or is there a particular exception you have in mind?

Reply:
Yeah just `raise _("Missing monitoring instasnce id") unles ...` is fine, similar to the missing ext_management_system raise above